cmd/cstorvolumeclaim/cstorvolumeclaim.go

```diff
@@ -342,26 +319,26 @@ func getOrCreateCStorVolumeResource(
 // on the available cstor pools created for storagepoolclaim.
 // if pools are less then desired replicaCount its return an error.
 func distributeCVRs(
-	replicaCount int,
+	desiredReplicaCount int,
+	claim *apis.CStorVolumeClaim,
 	service *corev1.Service,
 	volume *apis.CStorVolume,
-	class *storagev1.StorageClass,
 ) error {
 
-	spcName := getSPC(class)
+	spcName := getSPC(claim)
 	if len(spcName) == 0 {
-		return errors.New("failed to get spc name from storageClass")
+		return errors.New("failed to get spc name from cstorvolumeclaim")
 	}
 
-	poolList, err := listCStorPools(spcName, replicaCount)
+	poolList, err := listCStorPools(spcName, claim.Spec.ReplicaCount)
 	if err != nil {
 		return err
 	}
 
 	uniquePoolList := getUniquePoolList(volume.Name, poolList)
 	for i, pool := range uniquePoolList.Items {
 		pool := pool
-		if i < replicaCount {
+		if i < desiredReplicaCount {
```

Comment:
there are couple of issues happened here with old PR:
- `i` should be incremented only when createCVR doesn't return err
- at the end of this fn, if `i != desiredReplicaCount`, error should be returned
- if `i >= desiredReplicaCount`, break from the loop

review comment related to fn createCVR - it is just supposed to create CVR. If there exists one with the same name, we need to return error
i.e., there is no need of special handling in createCVR for `not existing` case

Reply:
As discussed this change is not required as we are already handling the pending replica creation during reconcile.